core/src/test/java/novoda/lib/sqliteprovider/provider/SQLiteProviderLocalTest.java

```diff
@@ -212,10 +281,24 @@ private void insert(String uri, ContentValues cv) {
         provider.insert(Uri.parse("content://" + uri), cv);
     }
 
+    private void bulkInsert(String uri, ContentValues[] cv) {
+        provider.bulkInsert(Uri.parse("content://" + uri), cv);
+    }
+
     private void query(String uri) {
         provider.query(Uri.parse("content://" + uri), null, null, null, null);
     }
 
+    private ContentValues[] createContentValuesArray(int size) {
+        ContentValues[] bulkToInsert = new ContentValues[size];
+        ContentValues contentValues = new ContentValues();
+        contentValues.put("test", "test");
+        for (int i = 0; i < size; i++) {
+            bulkToInsert[i] = contentValues;
+        }
+        return bulkToInsert;
+    }
+
     public class SQLiteProviderImpl extends SQLiteContentProviderImpl {
```

Comment:
It doesn't matter, exactly because you are using this stub as you would do with a mock: just to check if you are interacting with it, and in an atomic fashion (per single test) while focusing on testing **behaviour** instead of **state**.

My point is with a mock we can remove the state in this stub (that `notifyChangeCounter` is kinda gross to me) and have more expressive assertions, instead of depending on the internals of this stub.

> *Disclaimer*: I am more what Martin Fowler would define as ***"Mockist TDD practitioner"***
> [Mocks are not stubs](http://martinfowler.com/articles/mocksArentStubs.html)
> /cc @blundell @amlcurran